### PR TITLE
Add Mistral to launchdev.sh

### PR DESCRIPTION
**Pull request by Mierdin**
I was working on https://github.com/StackStorm/st2mistral/pull/30 and with the way things are installed on a "regular" system, this makes it difficult for dev. I realize others may have their own scripts for devving on Mistral, but I think it's worth adding to launchdev as an optional parameter.

So this PR does just that. For those that don't need the functionality, nothing changes - you have to explicitly pass in `-m` to use this.

For those that are interested, this will spin two `mistral-server` instances, one running the engine and executor, and the other running the api. Everything is done in screen as the other components, so if you specify `-m` you're just adding to that.

## Usage Example

```
(virtualenv)vagrant@st2dev:~/st2$ tools/launchdev.sh startclean -x -m && st2 run testpack.mistral-basic
Using st2 config file: /home/vagrant/st2/tools/../conf/st2.dev.conf
Using mistral config file: /home/vagrant/st2/tools/../conf/mistral.dev/mistral.dev.conf
Killing existing st2 screen sessions...
Killing existing mistral screen sessions...
Killing gunicorn processes
Initializing Mistral database. This will take some time because tox needs to download deps. Please be patient...
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files found matching '.gitreview'
warning: no previously-included files matching '*.pyc' found anywhere in distribution
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
No handlers could be found for logger "mistral.actions.openstack.action_generator.base"
Mistral database initialized.
MongoDB shell version: 2.4.9
connecting to: st2
[object Object]
Starting all st2 servers...
Changing working directory to /home/vagrant/st2/tools/.....
Using runners base dir: /opt/stackstorm/runners
Using config base dir: /opt/stackstorm/configs
Using content packs base dir: /opt/stackstorm/packs
Copying examples from ./contrib/examples to /opt/stackstorm/packs
No Sockets found in /var/run/screen/S-vagrant.

Starting screen session st2-api...
  using gunicorn to run st2-api...
  using gunicorn to run st2-stream
Starting screen sessions for st2-actionrunner(s)...
  starting st2-actionrunner-1...
Starting screen session st2-sensorcontainer
Starting screen session st2-rulesengine...
Starting screen session st2-resultstracker...
Starting screen session st2-notifier...
Starting screen session st2-auth...
  using gunicorn to run st2-auth...
Starting screen session mistral-server...
Starting screen session mistral-api...

Registering sensors, runners, actions, rules, aliases, and policies...
2017-07-11 02:52:25,557 INFO [-] Connecting to database "st2" @ "127.0.0.1:27017" as user "None".
2017-07-11 02:52:30,396 INFO [-] =========================================================
2017-07-11 02:52:30,403 INFO [-] ############## Registering triggers #####################
2017-07-11 02:52:30,403 INFO [-] =========================================================
2017-07-11 02:52:30,946 INFO [-] Registered 1 triggers.
2017-07-11 02:52:30,951 INFO [-] =========================================================
2017-07-11 02:52:30,951 INFO [-] ############## Registering sensors ######################
2017-07-11 02:52:30,951 INFO [-] =========================================================
2017-07-11 02:52:31,693 INFO [-] Registered 3 sensors.
2017-07-11 02:52:31,694 INFO [-] =========================================================
2017-07-11 02:52:31,695 INFO [-] ############## Registering runners ######################
2017-07-11 02:52:31,695 INFO [-] =========================================================
2017-07-11 02:52:33,778 INFO [-] Registered 13 runners.
2017-07-11 02:52:33,791 INFO [-] =========================================================
2017-07-11 02:52:33,792 INFO [-] ############## Registering actions ######################
2017-07-11 02:52:33,795 INFO [-] =========================================================
2017-07-11 02:52:39,608 INFO [-] Registered 100 actions.
2017-07-11 02:52:39,609 INFO [-] =========================================================
2017-07-11 02:52:39,610 INFO [-] ############## Registering rules ########################
2017-07-11 02:52:39,610 INFO [-] =========================================================
2017-07-11 02:52:40,283 INFO [-] Registered 8 rules.
2017-07-11 02:52:40,284 INFO [-] =========================================================
2017-07-11 02:52:40,284 INFO [-] ############## Registering aliases ######################
2017-07-11 02:52:40,284 INFO [-] =========================================================
2017-07-11 02:52:40,552 INFO [-] Registered 5 aliases.
2017-07-11 02:52:40,553 INFO [-] =========================================================
2017-07-11 02:52:40,555 INFO [-] ############## Registering policy types #################
2017-07-11 02:52:40,558 INFO [-] =========================================================
2017-07-11 02:52:40,712 INFO [-] Registered 3 policy types.
2017-07-11 02:52:40,714 INFO [-] =========================================================
2017-07-11 02:52:40,717 INFO [-] ############## Registering policies #####################
2017-07-11 02:52:40,725 INFO [-] =========================================================
2017-07-11 02:52:40,728 INFO [-] Registered 0 policies.
2017-07-11 02:52:40,728 INFO [-] =========================================================
2017-07-11 02:52:40,729 INFO [-] ############## Registering configs ######################
2017-07-11 02:52:40,729 INFO [-] =========================================================
2017-07-11 02:52:40,730 INFO [-] Registered 0 configs.
There are screens on:
	21670.mistral-api	(07/11/2017 02:52:05 AM)	(Detached)
	21666.mistral-server	(07/11/2017 02:52:05 AM)	(Detached)
	21664.st2-auth	(07/11/2017 02:52:05 AM)	(Detached)
	21661.st2-notifier	(07/11/2017 02:52:05 AM)	(Detached)
	21658.st2-resultstracker	(07/11/2017 02:52:05 AM)	(Detached)
	21655.st2-rulesengine	(07/11/2017 02:52:05 AM)	(Detached)
	21652.st2-sensorcontainer	(07/11/2017 02:52:05 AM)	(Detached)
	21648.st2-actionrunner-1	(07/11/2017 02:52:05 AM)	(Detached)
	21645.st2-stream	(07/11/2017 02:52:05 AM)	(Detached)
	21642.st2-api	(07/11/2017 02:52:05 AM)	(Detached)
10 Sockets in /var/run/screen/S-vagrant.

.....
id: 59643d7c32ed3554f94bf152
action.ref: testpack.mistral-basic
parameters: None
status: succeeded
result_task: task1
result:
  failed: false
  return_code: 0
  stderr: ''
  stdout: bar
  succeeded: true
start_timestamp: 2017-07-11T02:52:43.977577Z
end_timestamp: 2017-07-11T02:52:53.620184Z
+--------------------------+------------------------+-------+------------+-------------------------------+
| id                       | status                 | task  | action     | start_timestamp               |
+--------------------------+------------------------+-------+------------+-------------------------------+
| 59643d7f32ed3554f94bf155 | succeeded (1s elapsed) | task1 | core.local | Tue, 11 Jul 2017 02:52:47 UTC |
+--------------------------+------------------------+-------+------------+-------------------------------+
```

### CODE REVIEW

**Comment by armab** [comment]
Not a big fan of running mistral from sources in st2 repo, since this repository doesn't provide mistral source code. But if that works for you, - then yeah your idea of having a special flag to enable mistral makes sense, since we run all that launchev during the st2 integration tests CI where mistral is not needed.

**Comment by Mierdin** [comment]
Yeah, for sure it wouldn't be the default behavior, but I keep running into use cases where being able to have mistral in the launchdev screens would be super useful. So while Mistral isn't part of this repo, it's so core to a lot of what we do, I think it's good to at least have something optional in the dev script. I don't really have any scripts have for devving on Mistral so I figured I'd add to what I knew.

File: tools/launchdev.sh [hunk 1]
```diff
@@ -65,6 +68,76 @@ function init(){
...
+    st2stop
+    rm "${ST2_REPO}/mistral.db"
+
+    # Initialize mistral database if it doesn't already exist
+    if [ ! -f "${ST2_REPO}/mistral.db" ]; then
+        echo "Initializing Mistral database. This will take some time because tox needs to download deps. Please be patient..."
+
+        # We're using the StackStorm venv for everything
+        # TODO (mierdin): Evaluate if it's worth using a separate venv for installing and running mistral
+        source "${ST2_REPO}/virtualenv/bin/activate"
```
Comment (m4dcoder) [inline tools/launchdev.sh]: Can you put mistral requirements in its own venv? Please follow https://github.com/StackStorm/mistral_dev/blob/master/actions/install.sh.

File: tools/launchdev.sh [hunk 2]
```diff
@@ -65,6 +68,76 @@ function init(){
...
+        python setup.py install > /dev/null
+
+        # Tox is required by Mistral database initialization (will set up mistral/.tox)
+        pip install tox > /dev/null
+
+        # sync_db will run Tox, so we need to be in the mistral repo for that
+        cd "${MISTRAL_REPO}"
+
+        # Using mistral-db-manage doesn't seem to work with sqlite (tries to use ALTER) so we're using mistral's
+        # sync_db.sh script to do the table creation.
```
Comment (m4dcoder) [inline tools/launchdev.sh]: We don't support sqlite which doesn't work well w/ Mistral's transaction and multi-threading. Please use postgresql. Please follow https://github.com/StackStorm/mistral_dev/blob/master/actions/setup_db.sh.

**Comment by m4dcoder** [comment]
Please make sure that the mistral DB is handled properly for./launddev.sh start vs. startclean.

**Comment by Mierdin** [comment]
Alright, mistral is now installed into it's own virtualenv, it's using postgres, and start vs startclean seems to behave as desired. I'll do a bit more testing tomorrow to ensure the usual use cases are covered

**Comment by Kami** [comment]
LGTM, but still plan to test it :+1:

**Comment by Kami** [comment]
There appear to be some issues when running it multiple times (didn't really dig in), but besides that seems to work OK.

```bash
sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation "workbooks_v2" already exists
 [SQL: '\nCREATE TABLE workbooks_v2 (\n\tcreated_at TIMESTAMP WITHOUT TIME ZONE, \n\tupdated_at TIMESTAMP WITHOUT TIME ZONE, \n\tscope VARCHAR(80), \n\tproject_id VARCHAR(80), \n\tid VARCHAR(36) NOT NULL, \n\tname VARCHAR(80), \n\tdefinition TEXT, \n\tspec TEXT, \n\ttags TEXT, \n\tPRIMARY KEY (id), \n\tUNIQUE (name, project_id)\n)\n\n']
```

**Comment by Kami** [comment]
Looks like stop doesn't correctly kill mistral processes:

```bash
(virtualenv)vagrant@vagrant-ubuntu-trusty-64:/data/stanley$ ./tools/launchdev.sh stop -m
Killing existing st2 screen sessions...
Killing existing mistral screen sessions...
Killing gunicorn processes
(virtualenv)vagrant@vagrant-ubuntu-trusty-64:/data/stanley$ ps aux |grep mis
vagrant  10011  0.0  0.0  25772  1264 ?        Ss   09:41   0:00 SCREEN -d -m -S mistral-server /data/stanley/./tools/../../mistral/.venv/bin/python /data/stanley/./tools/../../mistral/.venv/bin/mistral-server --server engine,executor --config-file /data/stanley/./tools/../conf/mistral.dev/mistral.dev.conf --log-file /data/stanley/./tools/../logs/mistral-server.log
vagrant  10019  3.9  2.0 222760 83452 pts/15   Ss+  09:41   0:03 /data/stanley/./tools/../../mistral/.venv/bin/python /data/stanley/./tools/../../mistral/.venv/bin/mistral-server --server engine,executor --config-file /data/stanley/./tools/../conf/mistral.dev/mistral.dev.conf --log-file /data/stanley/./tools/../logs/mistral-server.log
postgres 10269  0.1  0.2 248644  8928 ?        Ss   09:42   0:00 postgres: mistral mistral 127.0.0.1(49739) idle                                                                             
postgres 10383  0.0  0.1 247900  7112 ?        Ss   09:43   0:00 postgres: mistral mistral 127.0.0.1(49825) idle                                                                             
vagrant  10476  0.0  0.0  10464   932 pts/7    S+   09:43   0:00 grep --color=auto mis
```

**Comment by Mierdin** [comment]
Yeah, I'm thinking this is related. Checking the processes a minute later shows everything shut down. Can add `-9`, at least for now.

**Comment by Mierdin** [comment]
@Kami Looks like adding the `-9` keeps things pretty clean:

```
vagrant@st2dev:~/st2$ tools/launchdev.sh startclean -x -m && sleep 1 && tools/launchdev.sh stop -m && ps aux |grep mis
Using st2 config file: /home/vagrant/st2/tools/../conf/st2.dev.conf
Initializing Mistral...
Using mistral config file: /home/vagrant/st2/tools/../conf/mistral.dev/mistral.dev.conf
Installing mistral, st2mistral, and all dependencies...
Requirement already satisfied (use --upgrade to upgrade): pytz in ./.venv/lib/python2.7/site-packages
Cleaning up...
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files found matching '.gitreview'
warning: no previously-included files matching '*.pyc' found anywhere in distribution
MongoDB shell version: 2.4.9
connecting to: st2
[object Object]
Killing gunicorn processes
Setup database in postgresql on ubuntu...
 * Restarting PostgreSQL 9.3 database server                                                                                                                                                                                 [ OK ]
DROP DATABASE
DROP ROLE
CREATE ROLE
CREATE DATABASE
Installing requirement libpg-dev...
Reading package lists... Done
Building dependency tree
Reading state information... Done
libpq-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 119 not upgraded.
Installing client lib for postgresql...
Creating tables...
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 001, Kilo release
INFO  [alembic.runtime.migration] Running upgrade 001 -> 002, Kilo
INFO  [alembic.runtime.migration] Running upgrade 002 -> 003, cron_trigger_constraints
INFO  [alembic.runtime.migration] Running upgrade 003 -> 004, add description for execution
INFO  [alembic.runtime.migration] Running upgrade 004 -> 005, Increase executions_v2 column size from JsonDictType to JsonLongDictType
INFO  [alembic.runtime.migration] Running upgrade 005 -> 006, add a Boolean column 'processed' to the table  delayed_calls_v2
INFO  [alembic.runtime.migration] Running upgrade 006 -> 007, Move system flag to base definition
INFO  [alembic.runtime.migration] Running upgrade 007 -> 008, Increase size of state_info column from String to Text
INFO  [alembic.runtime.migration] Running upgrade 008 -> 009, Add database indices
INFO  [alembic.runtime.migration] Running upgrade 009 -> 010, add_resource_members_v2_table
INFO  [alembic.runtime.migration] Running upgrade 010 -> 011, add workflow id for execution
INFO  [alembic.runtime.migration] Running upgrade 011 -> 012, add event triggers table
INFO  [alembic.runtime.migration] Running upgrade 012 -> 013, split_execution_table_increase_names
INFO  [alembic.runtime.migration] Running upgrade 013 -> 014, fix_past_scripts_discrepancies
INFO  [alembic.runtime.migration] Running upgrade 014 -> 015, add_unique_keys_for_non_locking_model
INFO  [alembic.runtime.migration] Running upgrade 015 -> 016, Increase size of task_executions_v2.unique_key
INFO  [alembic.runtime.migration] Running upgrade 016 -> 017, Add named lock table
INFO  [alembic.runtime.migration] Running upgrade 017 -> 018, increate_task_execution_unique_key_size
INFO  [alembic.runtime.migration] Running upgrade 018 -> 019, Change scheduler schema.
INFO  [alembic.runtime.migration] Running upgrade 019 -> 020, add type to task execution
Populating tables...
No handlers could be found for logger "oslo_db.sqlalchemy.engines"
========================
Database setup complete.
========================
Starting all st2 servers...
Changing working directory to /home/vagrant/st2/tools/.....
Using runners base dir: /opt/stackstorm/runners
Using config base dir: /opt/stackstorm/configs
Using content packs base dir: /opt/stackstorm/packs
Copying examples from ./contrib/examples to /opt/stackstorm/packs
No Sockets found in /var/run/screen/S-vagrant.

Starting screen session st2-api...
  using gunicorn to run st2-api...
  using gunicorn to run st2-stream
Starting screen sessions for st2-actionrunner(s)...
  starting st2-actionrunner-1...
Starting screen session st2-sensorcontainer
Starting screen session st2-rulesengine...
Starting screen session st2-resultstracker...
Starting screen session st2-notifier...
Starting screen session st2-auth...
  using gunicorn to run st2-auth...
Starting screen session mistral-server...
Starting screen session mistral-api...

Registering sensors, runners, actions, rules, aliases, and policies...
2017-07-24 18:59:38,793 INFO [-] Connecting to database "st2" @ "127.0.0.1:27017" as user "None".
2017-07-24 18:59:40,437 INFO [-] =========================================================
2017-07-24 18:59:40,438 INFO [-] ############## Registering triggers #####################
2017-07-24 18:59:40,438 INFO [-] =========================================================
2017-07-24 18:59:40,723 INFO [-] Registered 1 triggers.
2017-07-24 18:59:40,730 INFO [-] =========================================================
2017-07-24 18:59:40,730 INFO [-] ############## Registering sensors ######################
2017-07-24 18:59:40,730 INFO [-] =========================================================
2017-07-24 18:59:40,991 INFO [-] Registered 3 sensors.
2017-07-24 18:59:40,992 INFO [-] =========================================================
2017-07-24 18:59:40,992 INFO [-] ############## Registering runners ######################
2017-07-24 18:59:40,992 INFO [-] =========================================================
2017-07-24 18:59:41,678 INFO [-] Registered 13 runners.
2017-07-24 18:59:41,679 INFO [-] =========================================================
2017-07-24 18:59:41,679 INFO [-] ############## Registering actions ######################
2017-07-24 18:59:41,679 INFO [-] =========================================================
2017-07-24 18:59:44,489 INFO [-] Registered 103 actions.
2017-07-24 18:59:44,490 INFO [-] =========================================================
2017-07-24 18:59:44,490 INFO [-] ############## Registering rules ########################
2017-07-24 18:59:44,490 INFO [-] =========================================================
2017-07-24 18:59:44,715 INFO [-] Registered 8 rules.
2017-07-24 18:59:44,715 INFO [-] =========================================================
2017-07-24 18:59:44,716 INFO [-] ############## Registering aliases ######################
2017-07-24 18:59:44,716 INFO [-] =========================================================
2017-07-24 18:59:44,771 INFO [-] Registered 5 aliases.
2017-07-24 18:59:44,771 INFO [-] =========================================================
2017-07-24 18:59:44,771 INFO [-] ############## Registering policy types #################
2017-07-24 18:59:44,771 INFO [-] =========================================================
2017-07-24 18:59:44,794 INFO [-] Registered 3 policy types.
2017-07-24 18:59:44,794 INFO [-] =========================================================
2017-07-24 18:59:44,794 INFO [-] ############## Registering policies #####################
2017-07-24 18:59:44,794 INFO [-] =========================================================
2017-07-24 18:59:44,795 INFO [-] Registered 0 policies.
2017-07-24 18:59:44,795 INFO [-] =========================================================
2017-07-24 18:59:44,795 INFO [-] ############## Registering configs ######################
2017-07-24 18:59:44,795 INFO [-] =========================================================
2017-07-24 18:59:44,796 INFO [-] Registered 0 configs.
There are screens on:
	5883.mistral-api	(07/24/2017 06:59:34 PM)	(Detached)
	5880.mistral-server	(07/24/2017 06:59:34 PM)	(Detached)
	5877.st2-auth	(07/24/2017 06:59:33 PM)	(Detached)
	5874.st2-notifier	(07/24/2017 06:59:33 PM)	(Detached)
	5871.st2-resultstracker	(07/24/2017 06:59:33 PM)	(Detached)
	5868.st2-rulesengine	(07/24/2017 06:59:33 PM)	(Detached)
	5865.st2-sensorcontainer	(07/24/2017 06:59:33 PM)	(Detached)
	5861.st2-actionrunner-1	(07/24/2017 06:59:33 PM)	(Detached)
	5858.st2-stream	(07/24/2017 06:59:33 PM)	(Detached)
	5855.st2-api	(07/24/2017 06:59:33 PM)	(Detached)
10 Sockets in /var/run/screen/S-vagrant.

Killing existing st2 screen sessions...
Killing existing mistral screen sessions...
Killing gunicorn processes
vagrant   6059  0.0  0.0  10460   932 pts/0    S+   18:59   0:00 grep --color=auto mis
```

**Comment by nmaludy** [comment]
@Kami i'm revamping the puppet-st2 module and am running across a similar mistral startup error on my end.

This only occurs the first time i start the `mistral`(which starts `mistral-server` and `mistral-api`).  A simple restart brings the service back up without an issue.

Here are the logs i'm seeing (fairly consistently now):
```
 cat /var/log/mistral/mistral-api.log 
[2017-07-24 22:29:14 +0000] [11911] [INFO] Starting gunicorn 19.7.1
[2017-07-24 22:29:14 +0000] [11911] [INFO] Listening at: http://127.0.0.1:8989 (11911)
[2017-07-24 22:29:14 +0000] [11911] [INFO] Using worker: sync
[2017-07-24 22:29:14 +0000] [11948] [INFO] Booting worker with pid: 11948
[2017-07-24 22:29:14 +0000] [11949] [INFO] Booting worker with pid: 11949
[2017-07-24 22:29:18 +0000] [11948] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/arbiter.py", line 578, in spawn_worker
    worker.init_process()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/workers/base.py", line 135, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/util.py", line 352, in import_app
    __import__(module)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/api/wsgi.py", line 23, in <module>
    application = app.setup_app()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/api/app.py", line 54, in setup_app
    db_api_v2.setup_db()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/db/v2/api.py", line 29, in setup_db
    IMPL.setup_db()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/db/v2/sqlalchemy/api.py", line 68, in setup_db
    models.Workbook.metadata.create_all(b.get_engine())
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3762, in create_all
    tables=tables)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1856, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1481, in _run_visitor
    **kwargs).traverse_single(element)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 730, in visit_metadata
    _is_metadata_operation=True)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 764, in visit_table
    include_foreign_key_constraints=include_foreign_key_constraints
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 968, in _execute_ddl
    compiled
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1337, in _handle_dbapi_exception
    util.raise_from_cause(newraise, exc_info)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
DBError: (psycopg2.ProgrammingError) relation "workbooks_v2" already exists
 [SQL: '\nCREATE TABLE workbooks_v2 (\n\tscope VARCHAR(80), \n\tproject_id VARCHAR(80), \n\tcreated_at TIMESTAMP WITHOUT TIME ZONE, \n\tupdated_at TIMESTAMP WITHOUT TIME ZONE, \n\tid VARCHAR(36) NOT NULL, \n\tname VARCHAR(255), \n\tdefinition TEXT, \n\tspec TEXT, \n\ttags TEXT, \n\tis_system BOOLEAN, \n\tPRIMARY KEY (id), \n\tUNIQUE (name, project_id)\n)\n\n']
[2017-07-24 22:29:18 +0000] [11948] [INFO] Worker exiting (pid: 11948)
[2017-07-24 22:29:18 +0000] [11949] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/arbiter.py", line 578, in spawn_worker
    worker.init_process()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/workers/base.py", line 135, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/gunicorn/util.py", line 352, in import_app
    __import__(module)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/api/wsgi.py", line 23, in <module>
    application = app.setup_app()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/api/app.py", line 54, in setup_app
    db_api_v2.setup_db()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/db/v2/api.py", line 29, in setup_db
    IMPL.setup_db()
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/db/v2/sqlalchemy/api.py", line 68, in setup_db
    models.Workbook.metadata.create_all(b.get_engine())
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3762, in create_all
    tables=tables)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1856, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1481, in _run_visitor
    **kwargs).traverse_single(element)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 730, in visit_metadata
    _is_metadata_operation=True)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 121, in traverse_single
    return meth(obj, **kw)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 764, in visit_table
    include_foreign_key_constraints=include_foreign_key_constraints
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 968, in _execute_ddl
    compiled
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1337, in _handle_dbapi_exception
    util.raise_from_cause(newraise, exc_info)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/opt/stackstorm/mistral/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
DBDuplicateEntry: (psycopg2.IntegrityError) duplicate key value violates unique constraint "pg_type_typname_nsp_index"
DETAIL:  Key (typname, typnamespace)=(action_definitions_v2, 2200) already exists.
 [SQL: '\nCREATE TABLE action_definitions_v2 (\n\tscope VARCHAR(80), \n\tproject_id VARCHAR(80), \n\tcreated_at TIMESTAMP WITHOUT TIME ZONE, \n\tupdated_at TIMESTAMP WITHOUT TIME ZONE, \n\tid VARCHAR(36) NOT NULL, \n\tname VARCHAR(255), \n\tdefinition TEXT, \n\tspec TEXT, \n\ttags TEXT, \n\tis_system BOOLEAN, \n\tdescription TEXT, \n\tinput TEXT, \n\taction_class VARCHAR(200), \n\tattributes TEXT, \n\tPRIMARY KEY (id), \n\tUNIQUE (name, project_id)\n)\n\n']
[2017-07-24 22:29:18 +0000] [11949] [INFO] Worker exiting (pid: 11949)
```

**Comment by Kami** [comment]
@nmaludy Yeah, it's a bit weird - it looks like if mistral manage db script which runs DB migrations didn't run...

**Comment by Kami** [comment]
@Mierdin Yeah, -1 works for me.

**Comment by Kami** [comment]
LGTM and looks like you addressed all the comments :+1:

**Comment by Mierdin** [comment]
@Kami Np, and I assume you meant `-9` :smile:

**Pull request merged**